### ISSUE TITLE
`includeRoles` option in `control-has-associated-label` config is not implemented

### ISSUE THREAD

**Issue by nelsonlaidev** [jsx-eslint/eslint-plugin-jsx-a11y#1072]
The `recommended` and `strict` configurations for the `control-has-associated-label` rule include an `includeRoles` option:

```javascript
'jsx-a11y/control-has-associated-label': [
  'off',
  {
    // ... other options
    includeRoles: ['alert', 'dialog'],
  },
]
```

However, this option is not implemented in the actual rule. The rule only supports the following options:
- `labelAttributes`
- `controlComponents`
- `ignoreElements`
- `ignoreRoles`
- `depth`

The `includeRoles` option does not appear in:
- The rule implementation ([control-has-associated-label.js](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/8f75961d965e47afb88854d324bd32fafde7acfe/src/rules/control-has-associated-label.js))
- The rule documentation ([control-has-associated-label.md](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/8f75961d965e47afb88854d324bd32fafde7acfe/docs/rules/control-has-associated-label.md))
- The schema [definition](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/8f75961d965e47afb88854d324bd32fafde7acfe/src/rules/control-has-associated-label.js#L28-L38)

**Expected behavior:**
Either:
1. Remove `includeRoles` from the recommended/strict configurations since it has no effect, or
2. Implement the `includeRoles` option in the rule to explicitly include certain roles that should require labels (opposite of `ignoreRoles`)

**Location:**
- index.js (recommended config)
https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/8f75961d965e47afb88854d324bd32fafde7acfe/src/index.js#L84

- index.js (strict config)
https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/8f75961d965e47afb88854d324bd32fafde7acfe/src/index.js#L238

**Comment by rdcrampton** [comment]
I've put together a fix for this — PR incoming shortly.

The implementation adds `includeRoles` to the rule's schema, destructures it from options, and adds a check so that elements with roles in `includeRoles` are required to have an accessible label. This means the existing `includeRoles: ['alert', 'dialog']` in the recommended and strict configs will now actually be enforced.

Key design decisions:
- `ignoreRoles` takes precedence over `includeRoles` (checked first via early return)
- `includeRoles` is not gated behind `isDOMElement`, so custom components like `<MyAlert role="alert" />` also require labels
- Hidden elements (`aria-hidden="true"`) are still excluded
